### PR TITLE
Add FileNameStar to MultipartFileContent in WebCmdlets

### LINKED ISSUE #19456: Invoke-RestMethod Multipart/Form-Data Filename encoding with non-ascii chars
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

I am using the invoke-restmethod to do a Multipart/Form-Data Submission using the -Form parameter.
The form date also includes a file.
`$DocumentUploadForm = @{`
           ` file        = Get-Item $DocumentFile`
            `title       = "$DocumentTitle"`
            
When the filename contains just ascii chars the filename is encoded as expected in the request body.
However when the filename contains non-ascii chars the filename is encoded as per RFC-1342 which is unexpected.
I have compared the behavior to the CURL application and there the filename is encoded differently using unicode codepoints.

### Expected behavior

```console
Request Body for file with non-ascci chars in the filename:

...
--------------------------748339ec18b72df1
Content-Disposition: form-data; name="file"; filename="t\xcc\x88stfile.pdf"
Content-Type: application/pdf

%PDF-1.7
...

Request Body for file with only ascii chars in the filename:

...
--------------------------748339ec18b72df1
Content-Disposition: form-data; name="file"; filename="testfile.pdf"
Content-Type: application/pdf

%PDF-1.7
...
```

### Actual behavior

```console
Request Body for file with non-ascci chars in the filename:

...
--------------------------748339ec18b72df1
Content-Disposition: form-data; name="file"; filename="=?utf-8?B?S3XMiG5kaWd1bmdfTXVzdGVybWFubl9NYXguVFRBXzIwMjNfMDFfMzAucGRm?="
Content-Type: application/pdf

%PDF-1.7
...
```

### Error details

```console
There is no error in the powershell, but the server does not accept the POST request containing this filename encoding.
The server does accept a the POST request created by CURL.
The only difference is the encoding auf the filename.
Either with unicode codepoints or with RFC-1342
```

### Environment data

```powershell
Name                           Value
----                           -----
PSVersion                      7.3.3
PSEdition                      Core
GitCommitId                    7.3.3
OS                             Darwin 22.4.0 Darwin Kernel Version 22.4.0: Mon Mar  6 20:59:58 PST 2023; root:xnu-8796.101.5~3/RELEASE_ARM64_T6020
Platform                       Unix
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

_No response_

### PR DESCRIPTION
**Pull request by CarloToso**
# PR Summary

1. Add FileNameStar to MultipartFileContent

- The parameters filename and filename* differ only in that filename* uses the encoding defined in [RFC 5987](https://datatracker.ietf.org/doc/html/rfc5987). When both filename and filename* are present in a single header field value, filename* is preferred over filename when both are understood [from [https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Disposition]()]

2. Now .Net add quotes as needed so remove the code.

## PR Context
Removed linked issue as this PR doesn't fix that issue

## PR Checklist

- [x] [PR has a meaningful title](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
    - Use the present tense and imperative mood when describing your changes
- [x] [Summarized changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] [Make sure all `.h`, `.cpp`, `.cs`, `.ps1` and `.psm1` files have the correct copyright header](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
- [x] This PR is ready to merge and is not [Work in Progress](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---work-in-progress).
    - If the PR is work in progress, please add the prefix `WIP:` or `[ WIP ]` to the beginning of the title (the `WIP` bot will keep its status check at `Pending` while the prefix is present) and remove the prefix when the PR is ready.
- **[Breaking changes](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#making-breaking-changes)**
    - [x] None
    - **OR**
    - [ ] [Experimental feature(s) needed](https://github.com/MicrosoftDocs/PowerShell-Docs/blob/main/reference/7.3/Microsoft.PowerShell.Core/About/about_Experimental_Features.md)
        - [ ] Experimental feature name(s): 
- **User-facing changes**
    - [ ] Not Applicable
    - **OR**
    - [ ] [Documentation needed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#pull-request---submission)
        - [ ] Issue filed: 
- **Testing - New and feature**
    - [x] N/A or can only be tested interactively
    - **OR**
    - [ ] [Make sure you've added a new test if existing tests do not effectively test the code changed](https://github.com/PowerShell/PowerShell/blob/master/.github/CONTRIBUTING.md#before-submitting)
- **Tooling**
    - [ ] I have considered the user experience from a tooling perspective and don't believe tooling will be impacted.
    - **OR**
    - [ ] I have considered the user experience from a tooling perspective and opened an issue in the relevant tool repository. This may include:
        - [ ] Impact on [PowerShell Editor Services](https://github.com/PowerShell/PowerShellEditorServices) which is used in the [PowerShell extension](https://github.com/PowerShell/vscode-powershell) for VSCode
        (which runs in a different PS Host).
            - [ ] Issue filed: 
        - [ ] Impact on Completions (both in the console and in editors) - one of PowerShell's most powerful features.
            - [ ] Issue filed: 
        - [ ] Impact on [PSScriptAnalyzer](https://github.com/PowerShell/PSScriptAnalyzer) (which provides linting & formatting in the editor extensions).
            - [ ] Issue filed: 
        - [ ] Impact on [EditorSyntax](https://github.com/PowerShell/EditorSyntax) (which provides syntax highlighting with in VSCode, GitHub, and many other editors).
            - [ ] Issue filed:

### CODE REVIEW

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 1]
```diff
@@ -1747,6 +1747,7 @@ private static StreamContent GetMultipartFileContent(object fieldName, FileInfo
 
             // .NET does not enclose field names in quotes, however, modern browsers and curl do.
             result.Headers.ContentDisposition.FileName = "\"" + file.Name + "\"";
+            result.Headers.ContentDisposition.FileNameStar = "\"" + file.Name + "\"";
```
Comment (iSazonov) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1750]: I believe we should remove quotes. Now .Net does right things for both fields.

**Review by iSazonov — APPROVED** [review]

**Comment by daxian-dbw** [comment]
@CarloToso Can you please update the PR description with more context information about why adding `FileNameStar`? Also, can you please confirm whether it fixes https://github.com/PowerShell/PowerShell/issues/19456?

**Comment by CarloToso** [comment]
@daxian-dbw Updated the PR description, I can't confirm it fixes the issue because it depends on the endpoint that recieves the Multipart/Form-Data

**Comment by daxian-dbw** [comment]
Thanks @CarloToso. Let's wait for the user to verify the fix.

**Comment by TravisEz13** [comment]
This did not fix the customer's issue.  Do we still want to proceed with this fix?

**Comment by CarloToso** [comment]
@TravisEz13 I think we should merge this PR anyway

**Comment by iSazonov** [comment]
This add useful standard behavior.

**Comment by TravisEz13** [comment]
I pinged WG-Cmdlets to triage.

**Comment by SteveL-MSFT** [comment]
Looking at the HTTP specs, it seems that this is useful even though it doesn't solve the original problem.

**Review by SteveL-MSFT — APPROVED** [review]

**Comment by JamesWTruher** [comment]
if we're adding new or altering current behavior, there should be some validation.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 2]
```diff
@@ -1745,8 +1741,8 @@ private static StreamContent GetMultipartFileContent(object fieldName, FileInfo
         {
             StreamContent result = GetMultipartStreamContent(fieldName: fieldName, stream: new FileStream(file.FullName, FileMode.Open));
 
-            // .NET does not enclose field names in quotes, however, modern browsers and curl do.
-            result.Headers.ContentDisposition.FileName = "\"" + file.Name + "\"";
+            result.Headers.ContentDisposition.FileName = file.Name;
+            result.Headers.ContentDisposition.FileNameStar = file.Name;
```
Comment (JamesWTruher) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1745]: I think we should have tests for this new behavior.

**Review by JamesWTruher — CHANGES_REQUESTED** [review]
This really should have some validation of the behavior.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs (hunk 2 above)
Comment (CarloToso) [reply src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1745]: Done

**Comment by SteveL-MSFT** [comment]
@PowerShell/wg-powershell-cmdlets discussed this briefly and agree on supporting FileNameStar

**Comment by doctordns** [comment]
The Cmdlet working group has reviewed this proposal and agreed to its implementation.

**Review by SteveL-MSFT — APPROVED** [review]
Tests look good to me

**Review by JamesWTruher — APPROVED** [review]
thanks for adding the validation!

**Comment by marekvinkler** [comment]
It is possible this change is breaking `Invoke-RestMethod` calls with multipart/form-data, i.e.
```
$form = @{
  "file"      = Get-Item -Path $File
}
Invoke-RestMethod -Uri $Url -Method POST -Headers $headers -Form $form
```
results in an malformed request.

File: src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs [hunk 3]
```diff
@@ -1707,9 +1707,7 @@ private void AddMultipartContent(object fieldName, object fieldValue, MultipartF
         private static StringContent GetMultipartStringContent(object fieldName, object fieldValue)
         {
             ContentDispositionHeaderValue contentDisposition = new("form-data");
-
-            // .NET does not enclose field names in quotes, however, modern browsers and curl do.
-            contentDisposition.Name = "\"" + LanguagePrimitives.ConvertTo<string>(fieldName) + "\"";
+            contentDisposition.Name = LanguagePrimitives.ConvertTo<string>(fieldName);
```
Comment (rkeithhill-keysight) [inline src/Microsoft.PowerShell.Commands.Utility/commands/utility/WebCmdlet/Common/WebRequestPSCmdlet.Common.cs:1710]: This broke IWR for us when we upgraded to 7.4.  We use IWR in a publish script to upload a zip to an internal PERL / CGI-based website.  AFAICT the PERL CGI module requires field name values to be quoted.

**Pull request merged**